shared/oae/api/oae.api.tenancysearch.js

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright 2017 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+define(['jquery', 'oae.core', 'underscore', 'select2'], function($, oae, _, select2) {
+
+    let location = $.url(encodeURI(window.location.href));
+    let protocol = location.attr('protocol');
+    let query = location.attr('relative');
+
+    return (function () {
+        'use strict';
+
+        let searchTenancyModule =  {
+            init: function() {
+                this.cacheModule();
+                this.bindEvents();
+                this.render();
+            },
+            cacheModule: function() {
+                this.$module = $('#tenancyForm');
+                this.$input = this.$module.find('#tenancyInput');
+                this.$alert = this.$module.find('#tenancyAlert');
+            },
+            bindEvents: function() {
+                this.$input.on('select2:select', this.tenancySelect.bind(this));
+            },
+            render: function() {
+                this.$input.select2({
+                    dropdownParent: $('#signin-modal'),
+                    ajax: {
+                        url: '/api/search/tenants?limit=2000',
```

Comment:
Will 2000 always be enough?

Reply:
how likely is it going to be having the search query resulting in over 2000 entries? 🤷‍♂️

Reply:
Pretty unlikely, I admit.

Reply:
well, the user will have to type in more than that. It's like googling "university" trying to find a specific one, except that this is just a dropdown picker thus cannot have proper pagination, only limited scroll... I think that the 2000 limit is more of a protection for the DB query more than anything else. Any ideas? 🤷‍♂️

Reply:
Yeah, obviously it'd be a bad query, but I can still totally see someone doing it... 🤷‍♀️ I also noticed that unity.ac is actually pretty bad with queries like that (try typing in 'university of' and then pausing for example). That's probably more a sorting problem though.

Reply:
plus we have this: `tenancyResults = tenancyResults.slice(0, 10);` ... which makes the 2000 kind of obsolete.

I think that it's implicit that the user needs to search something quite specific in order to find the intended tenancy. The display (dropdown picker) doesn't scale to a lot of results, but listing +2000 results isn't something we'd want to do either way - not even paged - cuz it would no longer be an improvement over what we have at the moment.

Reply:
It is, yes :)